src/web/demo.rkt

```diff
@@ -390,155 +517,101 @@
+      (define hash (sha1 (open-input-string formula-str)))
+      (semaphore-wait (run-local-error hash formula seed sample))
+      (hash-ref *completed-jobs* hash))))
 
-      (eprintf " complete\n")
-      (hasheq 'tree tree))))
+(define (run-alternatives hash formula seed sample)
+  (hash-set! *jobs* hash (*timeline*))
+  (define sema (make-semaphore))
+  (thread-send *worker-thread* (list 'alternatives hash formula sema seed sample))
+  sema)
 
 (define alternatives-endpoint
   (post-with-json-response
     (lambda (post-data)
-      (define formula (read-syntax 'web (open-input-string (hash-ref post-data 'formula))))
+      (define formula-str (hash-ref post-data 'formula))
+      (define formula (read-syntax 'web (open-input-string formula-str)))
       (define sample (hash-ref post-data 'sample))
-      (define seed (hash-ref post-data 'seed #f))
-      (eprintf "Alternatives job started on ~a..." formula)
-
-      (define test (parse-test formula))
-      (define vars (test-vars test))
-      (define repr (test-output-repr test))
-      (define pcontext (json->pcontext sample (test-context test)))
-      (define result (run-herbie 'alternatives test #:seed seed #:pcontext pcontext
-                                 #:profile? #f #:timeline-disabled? #t))
-      (match-define (list altns test-pcontext processed-pcontext) (job-result-backend result))
-      
-      (define splitpoints
-        (for/list ([alt altns]) 
-          (for/list ([var vars])
-            (define split-var? (equal? var (regime-var alt)))
-            (if split-var?
-                (for/list ([val (regime-splitpoints alt)])
-                  (real->ordinal (repr->real val repr) repr))
-                '()))))
-
-      (define fpcores
-        (for/list ([altn altns])
-          (~a (program->fpcore (alt-expr altn) (test-context test)))))
-  
-      (define histories
-        (for/list ([altn altns])
-          (let ([os (open-output-string)])
-            (parameterize ([current-output-port os])
-              (write-xexpr
-                `(div ([id "history"])
-                  (ol ,@(render-history altn
-                                        processed-pcontext
-                                        test-pcontext
-                                        (test-context test)))))
-              (get-output-string os)))))
-      (define derivations 
-        (for/list ([altn altns])
-                  (render-json altn
-                                        processed-pcontext
-                                        test-pcontext
-                                        (test-context test))))
-
-      (eprintf " complete\n")
-      (hasheq 'alternatives fpcores
-              'histories histories
-              'derivations derivations
-              'splitpoints splitpoints))))
+      (define seed (hash-ref post-data 'seed #f))    
+      (semaphore-wait (run-alternatives hash formula seed sample))
```

Comment:
Bug here with hash not being defined but I'm guessing calling the racket function with no arguments?

Reply:
I think you need to define the hash somewhere, much like the other endpoints

Reply:
Fixed in next PR